src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/RemoveAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -0,0 +1,159 @@
+            HelpMessage = "Connection monitor object.",
+            ParameterSetName = "SetByInputObject")]
+        [ValidateNotNull]
+        public PSConnectionMonitorResult InputObject { get; set; }
+
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByLocation")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            string connectionMonitorName = this.Name;
+            string resourceGroupName = this.ResourceGroupName;
+            string networkWatcherName = this.NetworkWatcherName;
+
+            if (ParameterSetName.Contains("SetByResourceId"))
+            {
+                ConnectionMonitorDetails connectionMonitorDetails = new ConnectionMonitorDetails();
+                connectionMonitorDetails = this.GetConnectionMonitorDetails(this.ResourceId);
+
+                connectionMonitorName = connectionMonitorDetails.ConnectionMonitorName;
+                resourceGroupName = connectionMonitorDetails.ResourceGroupName;
+                networkWatcherName = connectionMonitorDetails.NetworkWatcherName;
+            }
+            else if (ParameterSetName.Contains("SetByResource"))
+            {
+                resourceGroupName = this.NetworkWatcher.ResourceGroupName;
+                networkWatcherName = this.NetworkWatcher.Name;
+            }
+            else if (ParameterSetName.Contains("SetByInputObject"))
+            {
+                ConnectionMonitorDetails connectionMonitorDetails = new ConnectionMonitorDetails();
+                connectionMonitorDetails = this.GetConnectionMonitorDetails(this.InputObject.Id);
+
+                connectionMonitorName = connectionMonitorDetails.ConnectionMonitorName;
+                resourceGroupName = connectionMonitorDetails.ResourceGroupName;
+                networkWatcherName = connectionMonitorDetails.NetworkWatcherName;
+            }
+            else if (ParameterSetName.Contains("SetByLocation"))
+            {
+                var networkWatcher = this.GetNetworkWatcherByLocation(this.Location);
+
+                if (networkWatcher == null)
+                {
+                    throw new ArgumentException("There is no network watcher in the specified location");
```

Comment:
@irrogozh nit: for all `ArgumentException` being thrown for no network watcher in a provided location, I would change "...in the specified location" to "...in location {0}" and fill it in with the provided location

Reply:
Fixed